src/main/java/com/homeadvisor/kafdrop/KafDrop.java

```diff
@@ -107,19 +111,41 @@ public int getOrder()
          return Ordered.HIGHEST_PRECEDENCE + 10;
       }
 
-      @Override
-      public void onApplicationEvent(ApplicationEnvironmentPreparedEvent event)
-      {
-         final ConfigurableEnvironment environment = event.getEnvironment();
-         if (environment.containsProperty(SM_CONFIG_DIR))
-         {
-            Stream.of("kafdrop", "global")
-               .map(name -> readProperties(environment, name))
-               .filter(Objects::nonNull)
-               .forEach(iniPropSource -> environment.getPropertySources()
-                  .addBefore("applicationConfigurationProperties", iniPropSource));
-         }
-      }
+        @Override
+        public void onApplicationEvent(ApplicationEnvironmentPreparedEvent event)
+        {
+            final ConfigurableEnvironment environment = event.getEnvironment();
+
+            LOG.info("Initializing jaas config");
+            String env = environment.getProperty("kafka.env");
+            Boolean isSecured = environment.getProperty("kafka.isSecured", Boolean.class);
+            LOG.info("env: {} .Issecured kafka: {}", env, isSecured);
+            if (isSecured && Strings.isNullOrEmpty(env)) {
+                throw new RuntimeException("'env' cannot be null if connecting to secured kafka.");
+            }
+
+            LOG.info("ENV: {}", env);
+            String path;
+
+            if (isSecured) {
+                if ((env.equalsIgnoreCase("stage") || env.equalsIgnoreCase("prod") || env.equalsIgnoreCase("local"))) {
```

Comment:
Is there something special about these particular environment names?  Would rather not dictate how folks organize their environments.  There's already a check for a non-null environment, so couldn't this just use the environment name as provided?